yucca/planning/dataset_properties.py

```diff
@@ -56,66 +59,100 @@ def create_properties_pkl(data_dir, save_dir, suffix=".nii.gz"):
-                    spacings.append([1.0, 1.0, 1.0])
-
-                image = nifti_or_np_to_np(image)
-                image_msk = image[image > background_pixel_value]
-
-                means.append(np.mean(image_msk))
-                stds.append(np.std(image_msk))
-
-                maxmin = np.max([min, np.min(image_msk)])
-                min = np.min([min, np.min(image_msk)])
-                max = np.max([max, np.max(image_msk)])
-            except Exception as err:
-                logging.warn(
-                    f"Could not read `{subject}`, got error `{err}`."
-                    "Suppressing to finalize, but you might need to act accordingly."
-                )
-
-        intensity_results[mod_id] = {
-            "mean": float(np.mean(means)),
-            "min": float(min),
-            "max": float(max),
-            "maxmin": float(maxmin),
-            "std": float(np.mean(stds)),
-        }
+        # cross-modality metadata
+        sizes = sizes + metadata["sizes"]
+        spacings = spacings + metadata["spacings"]
 
     dims = [len(size) for size in sizes]
     assert all([dim == dims[0] for dim in dims]), f"not all volumes have the same number of dimensions. Sizes were: {dims}"
 
     properties["data_dimensions"] = int(len(sizes[0]))
     properties["original_sizes"] = sizes
     properties["original_spacings"] = spacings
-    properties["intensities"] = intensity_results
+    properties["intensities"] = intensities
     properties["original_max_size"] = np.max(sizes, 0).tolist()
     properties["original_min_size"] = np.min(sizes, 0).tolist()
     properties["original_median_size"] = np.median(sizes, 0).tolist()
     properties["original_max_spacing"] = np.max(spacings, 0).tolist()
     properties["original_min_spacing"] = np.min(spacings, 0).tolist()
     properties["original_median_spacing"] = np.median(spacings, 0).tolist()
+
     save_pickle(properties, join(save_dir, "dataset_properties.pkl"))
+
+
+def reduce(results):
+    means = []
+    mins = []
+    maxs = []
+    stds = []
+    spacings = []
+    sizes = []
+
+    for res in tqdm(results, desc="Reduce"):
+        means.append(res["mean"])
+        mins.append(res["min"])
+        maxs.append(res["max"])
+        stds.append(res["std"])
+        spacings.append(res["spacing"])
+        sizes.append(res["size"])
+
+    return {"means": means, "mins": mins, "maxs": maxs, "stds": stds, "spacings": spacings, "sizes": sizes}
+
+
+def process(subject, background_pixel_value=0):
+    try:
+        image = read_file_to_nifti_or_np(subject)
+        dim = len(image.shape)
+
+        if dim > 3:
+            logging.warn(
+                f"A volume has more than three dimensions. This is most often a mistake." f"Dims: {dim}, Vol: {subject}"
+            )
+
+        size = image.shape
+
+        if isinstance(image, nib.Nifti1Image):
+            spacing = get_nib_spacing(image).tolist()
+        else:
+            spacing = [1.0, 1.0, 1.0]
```

Comment:
(not your fault, my own that I just spotted now that you changed the code) change this to "[1.0] * dim" to also support 2D images